### PR TITLE
[goldilocks] Fix Ingress k8s API version

### PR DESCRIPTION
**Pull request by czunker**
**Why This PR?**

For k8s >= 1.19.0 it will create the Ingress object with networking.k8s.io/v1.
For version < 1.19.0 it will use networking.k8s.io/v1beta1.

**Changes**

**Checklist:**

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or helm-docs has been run for the charts that support it.

Signed-off-by: Christian Zunker <christian.zunker@codecentric.cloud>

### CODE REVIEW

**Comment by czunker** [comment]
I haven't really introduced new variables, but had to change the format for `paths`.
Before, it looked like this:
```
        paths:
          - /
```
Now, it has to look like this:
```
        paths:
          - path: /
            type: ImplementationSpecific
```
I think, the content of the `paths` list wasn't reflected in the docs, so I'm not quite sure how to handle this change. Any ideas?

File: stable/goldilocks/ci/ingress-values.yaml
Comment (sudermanjr) [inline stable/goldilocks/ci/ingress-values.yaml]: It doesn't seem necessary to add a new test file here. Please just modify the old one. What is added here doesn't conflict with that

File: stable/goldilocks/templates/dashboard-ingress.yaml
Comment (sudermanjr) [inline stable/goldilocks/templates/dashboard-ingress.yaml:4]: Awesome

File: stable/goldilocks/values.yaml
Comment (sudermanjr) [inline stable/goldilocks/values.yaml]: I don't think these need to be commented. In fact, I'm realizing this will actually be backward compatible if they are not

**Review by sudermanjr — CHANGES_REQUESTED** [review]

**Comment by czunker** [comment]
Thanks for your review. Should all be fixed now.

**Comment by AlexanderBabel** [comment]
@czunker Hey, it would be great if you also add support for `ingressClassName`. This field replacing the annotation `kubernetes.io/ingress.class`

https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation

Thank you in advance!

**Comment by czunker** [comment]
@AlexanderBabel Good point.
Because the field needs support from the installed ingress controller, it's up to the user to decide whether to use the filed or the annotation.

@sudermanjr Could you please review again?

**Comment by czunker** [comment]
The CI creates a different `helm-docs` output to the one generated on my machine.

Perhaps different versions? I use
```
helm-docs --version
helm-docs version 1.5.0
```

**Comment by lucasreed** [comment]
@czunker It looks like we are also using 1.5.0 in CI: https://github.com/FairwindsOps/charts/blob/master/.circleci/config.yml#L40

If you could rebase and fix the merge conflicts we'll get another review going, sorry for the delay!

**Comment by czunker** [comment]
@lucasreed Thanks for the update. Fixed the merge conflicts.

**Comment by lucasreed** [comment]
> @lucasreed Thanks for the update. Fixed the merge conflicts.

Thanks! Also, when you run helm-docs, are you using this command: `helm-docs --sort-values-order=file` ? That is what runs in CI to check for diffs.

**Comment by czunker** [comment]
Thanks @lucasreed 
That fixed it.

File: stable/goldilocks/values.yaml
Comment (lucasreed) [reply stable/goldilocks/values.yaml]: I agree, I think we can uncomment this and remove the `[]` from line 118

**Review by lucasreed — CHANGES_REQUESTED** [review]

File: stable/goldilocks/values.yaml
Comment (czunker) [reply stable/goldilocks/values.yaml]: Fixed

**Review by lucasreed — DISMISSED** [review]

**Comment by lucasreed** [comment]
Thanks for the contribution @czunker! LGTM, I'll let @sudermanjr look it over one more time as well.

File: stable/goldilocks/Chart.yaml
Comment (lucasreed) [inline stable/goldilocks/Chart.yaml]: Because of the possibility that someone would have breaking changes here if they are deploying currently with an ingress path definition that does not include `type` I think we should make this a major release version change to `4.0.0`

**Review by lucasreed — CHANGES_REQUESTED** [review]
Apologies for the back and forth here, but can you update the chart version (my inline comment) and also update the README with a note about upgrading to v4.0.0 that any ingress path definitions will require the `type` value to be defined now.

**Comment by czunker** [comment]
> Apologies for the back and forth here, but can you update the chart version (my inline comment) and also update the README with a note about upgrading to v4.0.0 that any ingress path definitions will require the `type` value to be defined now.

No Problem.

Bump the version and added a change note. Please have a look.

**Review by lucasreed — APPROVED** [review]
Thanks again! LGTM

**Comment by rust84** [comment]
Any chance that this will be merged soon?

**Comment by AlexanderBabel** [comment]
@rust84 If you are using the `subcharts` pattern, then you can just disable the ingress from this chart and define your ingress resource inside the `templates` directory. That's my current solution until this PR gets finally merged.

**Comment by sudermanjr** [comment]
Looks like we lost track of this. I updated the PR to resolve the conflicts. @lucasreed can you please double check this?

File: stable/goldilocks/Chart.yaml
Comment (sudermanjr) [inline stable/goldilocks/Chart.yaml:25]:
```suggestion
  condition: metrics-server.enabled

```

Comment (sudermanjr) [inline stable/goldilocks/Chart.yaml]:
```suggestion
  condition: metrics-server.enabled

```

File: stable/goldilocks/ci/test-values.yaml
Comment (sudermanjr) [inline stable/goldilocks/ci/test-values.yaml]:
```suggestion
```

**Review by lucasreed — APPROVED** [review]

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**
